### ISSUE TITLE
BMW changed authentication method (?)

### ISSUE THREAD

**Issue by tnowak** [TA2k/ioBroker.bmw#73]
Seems that BMW has changed their login mechanism and adapter can't authenticate anymore:

bmw.0 | 2024-11-20 21:21:25.929 | error | undefined
bmw.0 | 2024-11-20 21:21:25.928 | error | Start relogin in 5min
bmw.0 | 2024-11-20 21:21:25.927 | error | Please check username and password or too many logins in 5 minutes
bmw.0 | 2024-11-20 21:21:25.926 | error | {"error":"invalid_client","error_description":"Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)"}
bmw.0 | 2024-11-20 21:21:25.925 | error | AxiosError: Request failed with status code 401
bmw.0 | 2024-11-20 21:21:25.923 | error | Login failed

To reproduce, just restart your instance

**Comment by fresenbruch** [comment]
I have upgraded to 2.8.3 and the result was:

bmw.0 2024-11-21 19:26:39.536	error	{"error":"invalid_request","error_description":"The request is missing a required parameter, includes an unsupported parameter value (other than grant type), repeats a parameter, includes multiple credentials, utilizes more than one mechanism for authenticating the client, or is otherwise malformed"}

bmw.0 2024-11-21 19:26:39.535	error	AxiosError: Request failed with status code 400

bmw.0 2024-11-21 19:26:39.533	error	refresh token failed

There seams to be a major problem in the authentification process

**Comment by gBonfire** [comment]
2.8.3: Please generate a captcha in the instance settings.
I've do this and now it works.....

Is this a possible Solution for you too?

**Comment by fresenbruch** [comment]
I've done this several times but it does not work

**Comment by fresenbruch** [comment]
Solved: I have deleted and generated the instance again.

**Comment by Ralf-Niemeier** [comment]
Works fine.
Perfect Job !

Solved.

**Comment by Bender275** [comment]
It's not working for me yet.
I updated to 2.8.4 and also deleted and reinstalled the instance.
My problem: I can klick in "I'm a human", the box is working but stays unchecked. I can do this several times, but nothing changed.
More down at the line of Captcha, just "[object Object]" is entered.

BR, Bender275

**Comment by tnowak** [comment]
Thank you very much, Mr. Developer!
2.8.4 works fine for me.
Will this be one-time re-authentication or it will require re-captcha login every month or so?

**Comment by TA2k** [comment]
captcha is need only after firstadapter installation
@Bender275 Please check the login in the app

**Comment by TA2k** [comment]
I don't know I assume around 10min
But captcha is only needed after adapter installation or after deletion of the auth.session

**Comment by tipp88** [comment]
> I don't know I assume around 10min But captcha is only needed after adapter installation or after deletion of the auth.session Which 401 error exactly

2024-11-24 04:21:08.164 - error: bmw.0 (82140) refresh token failed
2024-11-24 04:21:08.164 - error: bmw.0 (82140) AxiosError: Request failed with status code 403
2024-11-24 04:21:08.164 - error: bmw.0 (82140) "403 Forbidden"
2024-11-24 04:21:08.164 - error: bmw.0 (82140) Start relogin in 1min
2024-11-24 04:22:08.282 - error: bmw.0 (82140) Login failed
2024-11-24 04:22:08.282 - error: bmw.0 (82140) AxiosError: Request failed with status code 401
2024-11-24 04:22:08.282 - error: bmw.0 (82140) {"error":"invalid_client","error_description":"Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)"}
2024-11-24 04:22:08.282 - error: bmw.0 (82140) Please check username and password or generate a new captcha in the instance settings
2024-11-24 04:22:08.282 - error: bmw.0 (82140) Please wait 5 minutes before trying again
2024-11-24 04:22:08.282 - error: bmw.0 (82140) Start relogin in 5min
2024-11-24 04:22:08.282 - error: bmw.0 (82140) undefined

**Comment by tipp88** [comment]
I had to manually delete auth.session and generate a new captcha in order to get it back operational.

**Comment by Bender275** [comment]
I don't get it working... :-(
I installed the app on a third device and get no Captcha, also after reinstalling the app.
Also in the browsers, no matter which browser, no matter which device - no captcha, login is always working with user and passwort.

I click on the cpatcha generation, but nothing happens.
Error in the log:
<html>
<body>

bmw.0 | 2024-11-24 20:03:17.968 | error | Please check username and password or generate a new captcha in the instance settings
-- | -- | -- | --
bmw.0 | 2024-11-24 20:03:17.968 | error | {"error":"invalid_client","error_description":"Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)"}
bmw.0 | 2024-11-24 20:03:17.966 | error | AxiosError: Request failed with status code 401
bmw.0 | 2024-11-24 20:03:17.964 | error | Login failed


</body>
</html>

I also deleted the instance and the adapter and reinstalled it again a few minutes ago - no success.
Where is the auth.session?

**Comment by tipp88** [comment]
> Where is the auth.session?

This the object:
![image](https://github.com/user-attachments/assets/59b3013b-0d1b-4a6b-9f7a-80016c1c3454)

**Comment by wattwetter** [comment]
The ident works here not with Safari but it works with chrome.

**Comment by Bender275** [comment]
I tried Firefox and Edge at 3 different Windows clients with and without private mode - I even created new user profiles.
I installed four different browsers at my smartphone.
I even tried it with the browser app of the TV.
I reinstalled the app at my tabled several times.
Of course, I disabled pihole every time - I even bypassed it by changing the DNS-server.
All without success - I don't get a captcha.

**Comment by TA2k** [comment]
You will not see a captcha. You check the checkbox and then click on submit and a long text string will appear then you press save

**Comment by Bender275** [comment]
![Screenshot 2024-11-27 183157](https://github.com/user-attachments/assets/ec522be4-3fa1-4686-8edd-bd870da582ed)
That's my problem: My checkbox remains empty, the string is [object Object] - I just don't get it :-(

**Comment by TA2k** [comment]
Please test the GitHub version and check the browser console with Developer Console via F12

**Comment by Ralf-Pfalz** [comment]
> Seems that BMW has changed their login mechanism and adapter can't authenticate anymore:
> 
> bmw.0 | 2024-11-20 21:21:25.929 | error | undefined bmw.0 | 2024-11-20 21:21:25.928 | error | Start relogin in 5min bmw.0 | 2024-11-20 21:21:25.927 | error | Please check username and password or too many logins in 5 minutes bmw.0 | 2024-11-20 21:21:25.926 | error | {"error":"invalid_client","error_description":"Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)"} bmw.0 | 2024-11-20 21:21:25.925 | error | AxiosError: Request failed with status code 401 bmw.0 | 2024-11-20 21:21:25.923 | error | Login failed
> 
> To reproduce, just restart your instance

Same here

**Comment by Ralf-Pfalz** [comment]
to install the adapter 2.8.4 via NPN and do the Captcha Auth solved this problem for me.

**Comment by jsieberg** [comment]
Same problem here. I tried multiple cases. With or Without Chapta. 
Delete and create new Instance solved the problem.

**Comment by TA2k** [comment]
what is the captcha text input showing in the settings

**Comment by Ralf-Pfalz** [comment]
> what is the captcha text input showing in the settings

There is also a Button.  Press it.

**Comment by jsieberg** [comment]
Correct. Mark the checkbox, Press the button below. A new Key will appear below. 
This will be used for authentication. 
works for me.

**Comment by Bender275** [comment]
> Please test the GitHub version and check the browser console with Developer Console via F12

I get the following message, when I click in the captcha:
XHRPOST
https://hcaptcha.com/getcaptcha/7244955f-8f30-4445-adff-4fefe059f815
[HTTP/3 403  121ms]

Das Cookie "__cflb" wurde abgelehnt, weil es sich in einem seitenübergreifenden Kontext befindet und sein "SameSite" auf "Lax" oder "Strict" gesetzt ist.

I don't know, what else I can configure in Firefox - everything is allowed or disabled.
The error also comes in all the other browsers, so I don't think of a configuration issue.

**Comment by Bender275** [comment]
![Screenshot 2024-11-30 104923](https://github.com/user-attachments/assets/b401c0f3-62fb-4cce-9ea5-e2c10fb2359b)

I installed Chrome on my laptop and mande the settings
- allow 3rd party cookies
- safe browsing: no security

**Comment by TA2k** [comment]
How the url looks like?

**Comment by Bender275** [comment]
Yes, this url and tha captcha behind it works for me.

And your hint was the solution!!!!!!
Usually, I connect to iobroker via tha NAME of the Raspi, not the IP.
So, I tested it via IP - and the captcha works!

Thank you very much for your patience and support!
